pkg/scaling/executor/scale_executor.go

```diff
@@ -35,6 +35,9 @@ import (
 const (
 	// Default cooldown period for a ScaleTarget if no cooldownPeriod is defined on the scaledObject
 	defaultCooldownPeriod = 5 * 60 // 5 minutes
+
+	// Cooldownperiod delay effective time
+	defaultInitialDelayCooldownPeriod = 5 * 60 // 5 minutes
```

Comment:
I'm not 100 sure about this, as it's a breaking change with current behaviour. I'd prefer keeping current behaviour as default

Reply:
Given it, if we have 0 as default to keep current behaviour, `InitialDelayCooldownPeriod` can be `int32` instead of `*int32`

Reply:
I will try to change *int32 to int32 and the default value to 0

Reply:
I have modified it, but I don't know how to write unit tests to describe this behavior

Reply:
I guess that an e2e test is better for testing this. I'd add a new test case under `internals` folder: https://github.com/kedacore/keda/tree/main/tests/internals
You can get more info about the e2e tests here: https://github.com/kedacore/keda/blob/main/tests/README.md

Reply:
I will take the time to add this e2e

Reply:
Alright, but I didn't notice any new unit tests that you added @JorTurFer

Reply:
I haven't add it xD I'd add it if I were you in that folder xD